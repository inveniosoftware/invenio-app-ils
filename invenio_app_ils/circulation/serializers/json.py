# -*- coding: utf-8 -*-
#
# Copyright (C) 2018-2020 CERN.
#
# invenio-app-ils is free software; you can redistribute it and/or modify it
# under the terms of the MIT License; see LICENSE file for more details.

"""Loan JSON serializers."""

from invenio_records_rest.serializers.json import JSONSerializer

from invenio_app_ils.literature.serializers.custom_fields import \
    field_cover_metadata

from .custom_fields import (field_is_overdue, field_pickup_location,
                            field_transaction_location, field_transaction_user)


class LoanJSONSerializer(JSONSerializer):
    """Serialize loan."""

    def transform_record(self, pid, record, links_factory=None, **kwargs):
        """Transform record into an intermediate representation."""
        loan = super().transform_record(
            pid, record, links_factory=links_factory, **kwargs
        )
        field_is_overdue(loan["metadata"])
        field_pickup_location(loan["metadata"])
        field_transaction_location(loan["metadata"])
        field_transaction_user(loan["metadata"])
        field_cover_metadata(loan["metadata"].get("document", {}))
        return loan

    def transform_search_hit(
        self, pid, record_hit, links_factory=None, **kwargs
    ):
        """Transform search result hit into an intermediate representation."""
        hit = super().transform_search_hit(
            pid, record_hit, links_factory=links_factory, **kwargs
        )
        field_is_overdue(hit["metadata"])
        field_cover_metadata(hit["metadata"].get("document", {}))
        return hit
