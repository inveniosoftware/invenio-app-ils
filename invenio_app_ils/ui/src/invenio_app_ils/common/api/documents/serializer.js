import isEmpty from 'lodash/isEmpty';
import { fromISO } from '../date';

function serializeSeries(series) {
  return {
    serial: series.filter(s => s.mode_of_issuance === 'SERIAL'),
    multipart: series.filter(s => s.mode_of_issuance === 'MULTIPART_MONOGRAPH'),
  };
}

function serializeResponse(hit) {
  let result = {};
  if (!isEmpty(hit)) {
    result['id'] = hit.id;
    result['created'] = fromISO(hit.created);
    result['updated'] = fromISO(hit.updated);
    if (hit.links) {
      result['links'] = hit.links;
    }
    if (!isEmpty(hit.metadata)) {
      result['metadata'] = hit.metadata;
      if (!isEmpty(hit.metadata.circulation.next_available_date)) {
        result['metadata']['circulation']['next_available_date'] = new Date(
          result['metadata']['circulation']['next_available_date']
        ).toDateString();
      }
      if (!isEmpty(hit.metadata.series)) {
        // Split the series based on mode of issuance - requested by the library
        result['metadata']['series'] = serializeSeries(hit.metadata.series);
      }
      result['pid'] = hit.metadata.pid;
    }
  }
  return result;
}

export const serializer = {
  fromJSON: serializeResponse,
};
