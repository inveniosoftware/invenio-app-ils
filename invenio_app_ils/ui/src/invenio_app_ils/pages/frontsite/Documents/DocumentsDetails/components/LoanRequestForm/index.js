import { connect } from 'react-redux';
import LoanRequestFormComponent from './LoanRequestForm';
import { requestLoanForDocument } from './state/actions';

const mapDispatchToProps = dispatch => ({
  requestLoanForDocument: (documentPid, optionalParams = {}) =>
    dispatch(requestLoanForDocument(documentPid, optionalParams)),
});

export const LoanRequestForm = connect(
  null,
  mapDispatchToProps
)(LoanRequestFormComponent);
