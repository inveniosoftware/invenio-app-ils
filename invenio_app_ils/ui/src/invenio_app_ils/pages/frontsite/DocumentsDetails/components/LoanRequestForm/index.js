import { connect } from 'react-redux';
import LoanRequestFormComponent from './LoanRequestForm';
import { requestLoanForDocument } from './state/actions';

const mapDispatchToProps = dispatch => ({
  requestLoanForDocument: (docPid, loanRequestData) =>
    dispatch(requestLoanForDocument(docPid, loanRequestData)),
});

export const LoanRequestForm = connect(
  null,
  mapDispatchToProps
)(LoanRequestFormComponent);
