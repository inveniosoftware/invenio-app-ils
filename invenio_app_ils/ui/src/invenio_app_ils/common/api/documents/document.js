import { http } from '../base';
import { serializer } from './serializer';

const documentURL = '/documents/';

const get = documentPid => {
  return http.get(`${documentURL}${documentPid}`).then(response => {
    response.data = serializer.fromJSON(response.data);
    return response;
  });
};

class QueryBuilder {
  constructor() {
    this.overbookedQuery = [];
    this.availableItemsQuery = [];
    this.withPendingLoansQuery = [];
  }

  overbooked() {
    this.overbookedQuery.push(`circulation.overbooked:true`);
    return this;
  }

  withAvailableItems() {
    this.availableItemsQuery.push('circulation.loanable_items:>0');
    return this;
  }

  withPendingLoans() {
    this.withPendingLoansQuery.push('circulation.pending_loans:>0');
    return this;
  }

  qs() {
    return this.overbookedQuery
      .concat(this.availableItemsQuery, this.withPendingLoansQuery)
      .join(' AND ');
  }
}

const queryBuilder = () => {
  return new QueryBuilder();
};

const list = query => {
  return http.get(`${documentURL}?q=${query}`).then(response => {
    const totalHits = response.data.hits.total;
    response.data = response.data.hits.hits.map(hit =>
      serializer.fromJSON(hit)
    );
    response.data.totalHits = totalHits;
    return response;
  });
};

const count = query => {
  return http.get(`${documentURL}?q=${query}`).then(response => {
    response.data = response.data.hits.total;
    return response;
  });
};

export const document = {
  get: get,
  list: list,
  count: count,
  query: queryBuilder,
  serializer: serializer,
  url: documentURL,
};
