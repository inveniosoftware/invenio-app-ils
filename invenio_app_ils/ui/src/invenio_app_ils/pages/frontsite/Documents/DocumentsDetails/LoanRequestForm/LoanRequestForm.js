import React, { Component } from 'react';
import { Button, Checkbox, Form } from 'semantic-ui-react';
import { Error } from '../../../../../common/components/Error';
import { DatePicker } from '../../../../../common/components';
import PropTypes from 'prop-types';
import { invenioConfig } from '../../../../../common/config';
import { DateTime } from 'luxon';
import { toShortDate } from '../../../../../common/api/date';
import isEmpty from 'lodash/isEmpty';

export default class LoanRequestForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      requestEndDate: '',
      deliveryMethod: '',
    };

    // init delivery method
    this.withDeliveryMethod = !isEmpty(
      invenioConfig.circulation.deliveryMethods
    );
    this.deliveryMethods = this.withDeliveryMethod
      ? Object.keys(invenioConfig.circulation.deliveryMethods).map(key => ({
          key: key,
          value: key,
          text: invenioConfig.circulation.deliveryMethods[key],
        }))
      : [];
    this.state['deliveryMethod'] = this.withDeliveryMethod
      ? this.deliveryMethods[1].value
      : null;
  }

  requestLoanButton = () => {
    return (
      <Button
        positive
        size="small"
        content="Request"
        onClick={this.requestLoan}
      />
    );
  };

  handleRequestEndDateChange = value => {
    this.setState({ requestEndDate: value });
  };

  handleDeliveryMethodChange = (_, { value }) => {
    this.setState({ deliveryMethod: value });
  };

  handleSubmit = () => {
    const documentPid = this.props.document.metadata.pid;
    const optionalParams = {};
    if (!isEmpty(this.state.requestEndDate)) {
      optionalParams.requestEndDate = this.state.requestEndDate;
    }
    if (!isEmpty(this.state.deliveryMethod)) {
      optionalParams.deliveryMethod = this.state.deliveryMethod;
    }
    this.props.requestLoanForDocument(documentPid, optionalParams);
  };

  renderDeliveryRadioButtons = () => {
    return this.deliveryMethods.map(method => (
      <Checkbox
        radio
        label={method.text}
        name={'deliveryMethodRadioGroup'}
        value={method.value}
        checked={this.state.deliveryMethod === method.value}
        onChange={this.handleDeliveryMethodChange}
        key={method.value}
      />
    ));
  };

  renderDeliveryMethodSelector = () => {
    return this.withDeliveryMethod ? (
      <>
        <Form.Field>
          <label>I would like to</label>
        </Form.Field>
        <Form.Field>{this.renderDeliveryRadioButtons()}</Form.Field>
      </>
    ) : null;
  };

  renderOptionalRequestExpirationDate = () => {
    const today = DateTime.local();
    const initialDate = new DateTime(today.plus({ days: 10 }));
    const max = new DateTime(
      today.plus({ days: invenioConfig.circulation.requestDuration })
    );
    return (
      <Form.Field>
        <label>Do you require it before a certain date?</label>
        <DatePicker
          initialDate={toShortDate(initialDate)}
          minDate={toShortDate(today)}
          maxDate={toShortDate(max)}
          placeholder="Pick the date"
          handleDateChange={this.handleRequestEndDateChange}
        />
      </Form.Field>
    );
  };

  render() {
    const { error } = this.props;
    return (
      <Error error={error}>
        <Form>
          {this.renderDeliveryMethodSelector()}
          {this.renderOptionalRequestExpirationDate()}
          <Form.Button fluid color="orange" onClick={this.handleSubmit}>
            Request
          </Form.Button>
        </Form>
      </Error>
    );
  }
}

LoanRequestForm.propTypes = {
  requestLoanForDocument: PropTypes.func.isRequired,
  document: PropTypes.object.isRequired,
};
