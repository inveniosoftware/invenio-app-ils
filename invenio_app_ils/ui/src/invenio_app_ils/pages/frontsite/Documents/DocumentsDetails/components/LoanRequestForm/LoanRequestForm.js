import React, { Component } from 'react';
import { Button, Form, Header, Segment } from 'semantic-ui-react';
import { Error } from '../../../../../../common/components/Error';
import { DatePicker } from '../../../../../../common/components';
import PropTypes from 'prop-types';
import { invenioConfig } from '../../../../../../common/config';
import { DateTime } from 'luxon';
import { toShortDate} from '../../../../../../common/api/date';
import isEmpty from 'lodash/isEmpty';

export default class LoanRequestForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      requestEndDate: '',
      deliveryMethod: '',
    };

    // init delivery method
    this.withDeliveryMethod = !isEmpty(
      invenioConfig.circulation.deliveryMethods
    );
    this.deliveryMethods = this.withDeliveryMethod
      ? Object.keys(invenioConfig.circulation.deliveryMethods).map(key => ({
          key: key,
          value: key,
          text: invenioConfig.circulation.deliveryMethods[key],
        }))
      : [];
    this.state['deliveryMethod'] = this.withDeliveryMethod
      ? this.deliveryMethods[1].value
      : null;
  }

  requestLoanButton = () => {
    return (
      <Button
        positive
        size="small"
        content="Request"
        onClick={this.requestLoan}
      />
    );
  };

  handleRequestEndDateChange = value => {
    this.setState({ requestEndDate: value });
  };

  handleDeliveryMethodChange = (_, { value }) => {
    this.setState({ deliveryMethod: value });
  };

  handleSubmit = () => {
    const documentPid = this.props.document.metadata.pid;
    const optionalParams = {};
    if (!isEmpty(this.state.requestEndDate)) {
      optionalParams.requestEndDate = this.state.requestEndDate;
    }
    if (!isEmpty(this.state.deliveryMethod)) {
      optionalParams.deliveryMethod = this.state.deliveryMethod;
    }
    this.props.requestLoanForDocument(documentPid, optionalParams);
  };

  renderDeliveryMethodSelector = () => {
    return this.withDeliveryMethod ? (
      <Form.Field>
        <label>Choose the book delivery method</label>
        <Form.Dropdown
          placeholder={'Select delivery method'}
          options={this.deliveryMethods}
          onChange={this.handleDeliveryMethodChange}
          defaultValue={this.deliveryMethods[1].value}
          selection
        />
      </Form.Field>
    ) : null;
  };

  renderOptionalRequestExpirationDate = () => {
    const today = DateTime.local();
    const initialDate = new DateTime(today.plus({ days: 10 }));
    const max = new DateTime(
      today.plus({ days: invenioConfig.circulation.requestDuration })
    );
    return (
      <div>
        <Segment.Inline>
          <div>Optionally, select a limit date for your request</div>
          <DatePicker
            initialDate={toShortDate(initialDate)}
            minDate={toShortDate(today)}
            maxDate={toShortDate(max)}
            placeholder="Request limit date"
            handleDateChange={this.handleRequestEndDateChange}
          />
        </Segment.Inline>
      </div>
    );
  };

  render() {
    const { error } = this.props;
    return (
      <Error error={error}>
        <Form>
          <Segment.Group>
            <Segment>
              <Header as="h3" content="Request loan" />
            </Segment>
            <Segment>
              {this.renderDeliveryMethodSelector()}
              {this.renderOptionalRequestExpirationDate()}
            </Segment>
            <Segment>
              <Form.Button onClick={this.handleSubmit}>Request</Form.Button>
            </Segment>
          </Segment.Group>
        </Form>
      </Error>
    );
  }
}

LoanRequestForm.propTypes = {
  requestLoanForDocument: PropTypes.func.isRequired,
  document: PropTypes.object.isRequired,
};
