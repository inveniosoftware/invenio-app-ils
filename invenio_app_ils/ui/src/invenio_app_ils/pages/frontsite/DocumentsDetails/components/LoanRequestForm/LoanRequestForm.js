import React, { Component } from 'react';
import { Button, Form, Header, Segment } from 'semantic-ui-react';
import { Error } from '../../../../../common/components/Error';
import { DateRangePicker } from '../../../../../common/components';
import PropTypes from 'prop-types';
import { sessionManager } from '../../../../../authentication/services';
import { invenioConfig } from '../../../../../common/config';
import { DateTime } from 'luxon';
import { toShortDate, toUTCShortDate } from '../../../../../common/api/date';

export default class LoanRequestForm extends Component {
  constructor(props) {
    super(props);
    const tomorrow = DateTime.local().plus({ days: 1 });
    const loanDuration = new DateTime(
      tomorrow.plus({ days: invenioConfig.circulation.defaultDuration })
    );
    this.state = {
      fromDate: props.defaultStartDate
        ? props.defaultStartDate
        : toShortDate(tomorrow),
      toDate: props.defaultEndDate
        ? props.defaultEndDate
        : toShortDate(loanDuration),
      deliveryMethod: this.deliveryMethods()[1].value,
    };
  }

  requestLoanButton = () => {
    return (
      <>
        <Button
          positive
          size="small"
          content="Loan"
          onClick={this.requestLoan}
        />
      </>
    );
  };

  handleDateChange = (name, value) => {
    this.setState({ [name]: value });
  };

  handleDeliveryMethodChange = (event, object) => {
    this.setState({ deliveryMethod: object.value });
  };

  handleSubmit = () => {
    const documentPid = this.props.document.metadata.pid;
    const loanRequestData = {
      metadata: {
        start_date: toUTCShortDate(this.state.fromDate),
        end_date: toUTCShortDate(this.state.toDate),
        document_pid: documentPid,
        patron_pid: sessionManager.user.id,
        delivery: {
          method: this.state.deliveryMethod,
        },
      },
    };
    this.props.requestLoanForDocument(documentPid, loanRequestData);
  };

  deliveryMethods = () => {
    return invenioConfig.circulation.deliveryMethods.map(method => {
      return { key: method, value: method, text: method };
    });
  };

  render() {
    const { error } = this.props;
    return (
      <Error error={error}>
        <Form>
          <Segment.Group>
            <Segment>
              <Header
                as="h3"
                content="Request loan"
                subheader="Choose the period of interest for the book"
              />
            </Segment>
            <Segment>
              <Segment.Group horizontal>
                <DateRangePicker
                  defaultStart={this.state.fromDate}
                  defaultEnd={this.state.toDate}
                  handleDateChange={this.handleDateChange}
                />
              </Segment.Group>
              <Form.Field>
                <label>Choose the book delivery method</label>
                <Form.Dropdown
                  placeholder={'Select delivery method'}
                  options={this.deliveryMethods()}
                  onChange={this.handleDeliveryMethodChange}
                  defaultValue={this.deliveryMethods()[1].value}
                  selection
                />
              </Form.Field>
              <Form.Button onClick={this.handleSubmit}>Request</Form.Button>
            </Segment>
          </Segment.Group>
        </Form>
      </Error>
    );
  }
}

LoanRequestForm.propTypes = {
  requestLoanForDocument: PropTypes.func.isRequired,
  document: PropTypes.object.isRequired,
  defaultStartDate: PropTypes.string,
  defaultEndDate: PropTypes.string,
};
