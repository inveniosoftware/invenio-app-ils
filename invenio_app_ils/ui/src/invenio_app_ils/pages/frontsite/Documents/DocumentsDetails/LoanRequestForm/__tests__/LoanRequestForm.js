import React from 'react';
import { mount } from 'enzyme';
import LoanRequestForm from '../LoanRequestForm';
import * as testData from '../../../../../../../../../../tests/data/documents.json';
import { DateTime } from 'luxon';

jest.mock('../../../../../../common/config/invenioConfig');

describe('DocumentMetadata tests', () => {
  let component;
  afterEach(() => {
    if (component) {
      component.unmount();
    }
  });

  const mockRequestLoanForDocument = jest.fn();

  const document = {
    id: 71,
    created: '2019-07-08T10:44:02.366+02:00',
    updated: '2019-07-08T10:44:18.354+02:00',
    links: { self: 'https://127.0.0.1:5000/api/documents/71' },
    metadata: {
      $schema: 'https://127.0.0.1:5000/schemas/documents/document-v1.0.0.json',
      _access: {},
      ...testData[0],
      relations: {},
      circulation: {},
      pid: '71',
      document_type: ['BOOK'],
      tag_pids: ['24'],
      tags: [],
      eitems: {
        total: 1,
        hits: [
          {
            description: 'Non ipsum',
            pid: '145',
            open_access: true,
          },
        ],
      },
    },
  };

  it('should render the loan request form correctly', () => {
    DateTime.local = jest
      .fn()
      .mockReturnValue(DateTime.fromObject({ year: 2019, month: 10, day: 1 }));
    component = mount(
      <LoanRequestForm
        document={document}
        requestLoanForDocument={mockRequestLoanForDocument}
      />
    );

    expect(component).toMatchSnapshot();

    const fields = component
      .find('LoanRequestForm')
      .find('DateInput')
      .filterWhere(element => element.prop('data-test') === '');
    expect(fields).toHaveLength(1);
  });
});
