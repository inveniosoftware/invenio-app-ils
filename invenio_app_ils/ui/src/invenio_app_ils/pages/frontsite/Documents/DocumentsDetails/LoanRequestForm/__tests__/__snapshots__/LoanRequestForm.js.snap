// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentMetadata tests should render the loan request form correctly 1`] = `
<LoanRequestForm
  document={
    Object {
      "created": "2019-07-08T10:44:02.366+02:00",
      "id": 71,
      "links": Object {
        "self": "https://127.0.0.1:5000/api/documents/71",
      },
      "metadata": Object {
        "$schema": "https://127.0.0.1:5000/schemas/documents/document-v1.0.0.json",
        "_access": Object {},
        "abstract": "This is an abstract",
        "authors": Array [
          Object {
            "full_name": "Jack E. Davis",
          },
        ],
        "circulation": Object {},
        "document_type": Array [
          "BOOK",
        ],
        "edition": "ed. 1",
        "eitems": Object {
          "hits": Array [
            Object {
              "description": "Non ipsum",
              "open_access": true,
              "pid": "145",
            },
          ],
          "total": 1,
        },
        "keywords": Object {
          "source": "X",
          "value": "Patata",
        },
        "languages": Array [
          "en",
        ],
        "pid": "71",
        "relations": Object {},
        "tag_pids": Array [
          "24",
        ],
        "tags": Array [],
        "title": "The Gulf: The Making of An American Sea",
      },
      "updated": "2019-07-08T10:44:18.354+02:00",
    }
  }
  requestLoanForDocument={[MockFunction]}
>
  <Error>
    <Form
      as="form"
    >
      <form
        className="ui form"
        onSubmit={[Function]}
      >
        <FormField
          required={true}
        >
          <div
            className="required field"
          >
            <label>
              I would like to
            </label>
          </div>
        </FormField>
        <FormField>
          <div
            className="field"
          >
            <Checkbox
              checked={true}
              key="DELIVERY"
              label=""
              name="deliveryMethodRadioGroup"
              onChange={[Function]}
              radio={true}
              type="checkbox"
              value="DELIVERY"
            >
              <div
                className="ui checked radio checkbox"
                onChange={[Function]}
                onClick={[Function]}
                onMouseDown={[Function]}
                onMouseUp={[Function]}
              >
                <Ref
                  innerRef={
                    Object {
                      "current": <input
                        checked=""
                        class="hidden"
                        name="deliveryMethodRadioGroup"
                        readonly=""
                        tabindex="0"
                        type="checkbox"
                        value="DELIVERY"
                      />,
                    }
                  }
                >
                  <RefFindNode
                    innerRef={
                      Object {
                        "current": <input
                          checked=""
                          class="hidden"
                          name="deliveryMethodRadioGroup"
                          readonly=""
                          tabindex="0"
                          type="checkbox"
                          value="DELIVERY"
                        />,
                      }
                    }
                  >
                    <input
                      checked={true}
                      className="hidden"
                      name="deliveryMethodRadioGroup"
                      readOnly={true}
                      tabIndex={0}
                      type="checkbox"
                      value="DELIVERY"
                    />
                  </RefFindNode>
                </Ref>
                <Ref
                  innerRef={
                    Object {
                      "current": <label />,
                    }
                  }
                >
                  <RefFindNode
                    innerRef={
                      Object {
                        "current": <label />,
                      }
                    }
                  >
                    <label />
                  </RefFindNode>
                </Ref>
              </div>
            </Checkbox>
            <Checkbox
              checked={false}
              key="PICK UP"
              label=""
              name="deliveryMethodRadioGroup"
              onChange={[Function]}
              radio={true}
              type="checkbox"
              value="PICK UP"
            >
              <div
                className="ui radio checkbox"
                onChange={[Function]}
                onClick={[Function]}
                onMouseDown={[Function]}
                onMouseUp={[Function]}
              >
                <Ref
                  innerRef={
                    Object {
                      "current": <input
                        class="hidden"
                        name="deliveryMethodRadioGroup"
                        readonly=""
                        tabindex="0"
                        type="checkbox"
                        value="PICK UP"
                      />,
                    }
                  }
                >
                  <RefFindNode
                    innerRef={
                      Object {
                        "current": <input
                          class="hidden"
                          name="deliveryMethodRadioGroup"
                          readonly=""
                          tabindex="0"
                          type="checkbox"
                          value="PICK UP"
                        />,
                      }
                    }
                  >
                    <input
                      checked={false}
                      className="hidden"
                      name="deliveryMethodRadioGroup"
                      readOnly={true}
                      tabIndex={0}
                      type="checkbox"
                      value="PICK UP"
                    />
                  </RefFindNode>
                </Ref>
                <Ref
                  innerRef={
                    Object {
                      "current": <label />,
                    }
                  }
                >
                  <RefFindNode
                    innerRef={
                      Object {
                        "current": <label />,
                      }
                    }
                  >
                    <label />
                  </RefFindNode>
                </Ref>
              </div>
            </Checkbox>
          </div>
        </FormField>
        <FormField>
          <div
            className="field"
          >
            <label>
              Do you require it before a certain date? (optional)
            </label>
            <DatePicker
              defaultValue=""
              handleDateChange={[Function]}
              initialDate="2019-10-11"
              maxDate="2019-11-30"
              minDate="2019-10-01"
              name="selectedDate"
              placeholder="Pick the date"
            >
              <DateInput
                autoComplete="off"
                clearable={true}
                closable={true}
                data-test=""
                dateFormat="DD-MM-YYYY"
                icon="calendar"
                iconPosition="left"
                initialDate="2019-10-11"
                inline={false}
                localization="en"
                maxDate="2019-11-30"
                minDate="2019-10-01"
                name="selectedDate"
                onChange={[Function]}
                placeholder="Pick the date"
                preserveViewMode={true}
                startMode="day"
                value=""
              >
                <InputView
                  animation="scale"
                  autoComplete="off"
                  clearIcon="remove"
                  clearable={true}
                  closeOnMouseLeave={true}
                  closePopup={[Function]}
                  data-test=""
                  duration={200}
                  icon="calendar"
                  iconPosition="left"
                  inline={false}
                  name="selectedDate"
                  onChange={[Function]}
                  onFocus={[Function]}
                  onMount={[Function]}
                  openPopup={[Function]}
                  placeholder="Pick the date"
                  popupIsClosed={true}
                  renderPicker={[Function]}
                  tabIndex="0"
                  value=""
                >
                  <FormInputWithRef
                    autoComplete="off"
                    clearIcon="remove"
                    clearable={true}
                    data-test=""
                    icon="calendar"
                    iconPosition="left"
                    innerRef={[Function]}
                    name="selectedDate"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onClear={[Function]}
                    onFocus={[Function]}
                    onMouseEnter={[Function]}
                    placeholder="Pick the date"
                    tabIndex="0"
                    value=""
                  >
                    <FormInput
                      as={[Function]}
                      autoComplete="off"
                      control={[Function]}
                      data-test=""
                      icon={true}
                      iconPosition="left"
                      name="selectedDate"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onMouseEnter={[Function]}
                      placeholder="Pick the date"
                      tabIndex="0"
                    >
                      <FormField
                        autoComplete="off"
                        control={[Function]}
                        data-test=""
                        icon={true}
                        iconPosition="left"
                        name="selectedDate"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onMouseEnter={[Function]}
                        placeholder="Pick the date"
                        tabIndex="0"
                      >
                        <div
                          className="field"
                        >
                          <Input
                            autoComplete="off"
                            data-test=""
                            icon={true}
                            iconPosition="left"
                            name="selectedDate"
                            onBlur={[Function]}
                            onChange={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            onMouseEnter={[Function]}
                            placeholder="Pick the date"
                            tabIndex="0"
                            type="text"
                          >
                            <div
                              className="ui left icon input"
                              data-test=""
                            >
                              <Icon
                                as="i"
                                key=".0"
                                name="calendar"
                              >
                                <i
                                  aria-hidden="true"
                                  className="calendar icon"
                                  onClick={[Function]}
                                />
                              </Icon>
                              <input
                                autoComplete="off"
                                key=".1"
                                name="selectedDate"
                                onBlur={[Function]}
                                onChange={[Function]}
                                onClick={[Function]}
                                onFocus={[Function]}
                                onMouseEnter={[Function]}
                                placeholder="Pick the date"
                                tabIndex="0"
                                type="text"
                                value=""
                              />
                            </div>
                          </Input>
                        </div>
                      </FormField>
                    </FormInput>
                  </FormInputWithRef>
                  <Transition
                    animation="scale"
                    duration={200}
                    mountOnShow={true}
                    onComplete={[Function]}
                    transitionOnMount={false}
                    unmountOnHide={true}
                    visible={false}
                  />
                </InputView>
              </DateInput>
            </DatePicker>
          </div>
        </FormField>
        <FormButton
          as={[Function]}
          control={[Function]}
          fluid={true}
          onClick={[Function]}
          primary={true}
        >
          <FormField
            control={[Function]}
            fluid={true}
            onClick={[Function]}
            primary={true}
          >
            <div
              className="field"
            >
              <Button
                as="button"
                fluid={true}
                onClick={[Function]}
                primary={true}
              >
                <Ref
                  innerRef={
                    Object {
                      "current": <button
                        class="ui fluid primary button"
                      >
                        Request
                      </button>,
                    }
                  }
                >
                  <RefFindNode
                    innerRef={
                      Object {
                        "current": <button
                          class="ui fluid primary button"
                        >
                          Request
                        </button>,
                      }
                    }
                  >
                    <button
                      className="ui fluid primary button"
                      onClick={[Function]}
                    >
                      Request
                    </button>
                  </RefFindNode>
                </Ref>
              </Button>
            </div>
          </FormField>
        </FormButton>
      </form>
    </Form>
  </Error>
</LoanRequestForm>
`;
